servicePretix.go

```diff
@@ -0,0 +1,177 @@
+			GivenName  string `json:"given_name"`
+			FamilyName string `json:"family_name"`
+		} `json:"attendee_name_parts"`
+		Company       any    `json:"company"`
+		Street        any    `json:"street"`
+		Zipcode       any    `json:"zipcode"`
+		City          any    `json:"city"`
+		Country       any    `json:"country"`
+		State         any    `json:"state"`
+		Discount      any    `json:"discount"`
+		AttendeeEmail any    `json:"attendee_email"`
+		Voucher       any    `json:"voucher"`
+		TaxRate       string `json:"tax_rate"`
+		TaxValue      string `json:"tax_value"`
+		Secret        string `json:"secret"`
+		AddonTo       any    `json:"addon_to"`
+		Subevent      any    `json:"subevent"`
+		Checkins      []any  `json:"checkins"`
+		Downloads     []struct {
+			Output string `json:"output"`
+			URL    string `json:"url"`
+		} `json:"downloads"`
+		Answers            []any  `json:"answers"`
+		TaxRule            any    `json:"tax_rule"`
+		PseudonymizationID string `json:"pseudonymization_id"`
+		Seat               any    `json:"seat"`
+		Canceled           bool   `json:"canceled"`
+		ValidFrom          any    `json:"valid_from"`
+		ValidUntil         any    `json:"valid_until"`
+		Blocked            any    `json:"blocked"`
+	} `json:"positions"`
+	Downloads []struct {
+		Output string `json:"output"`
+		URL    string `json:"url"`
+	} `json:"downloads"`
+	CheckinAttention bool      `json:"checkin_attention"`
+	CheckinText      any       `json:"checkin_text"`
+	LastModified     time.Time `json:"last_modified"`
+	Payments         []struct {
+		LocalID     int       `json:"local_id"`
+		State       string    `json:"state"`
+		Amount      string    `json:"amount"`
+		Created     time.Time `json:"created"`
+		PaymentDate any       `json:"payment_date"`
+		Provider    string    `json:"provider"`
+		PaymentURL  any       `json:"payment_url"`
+		Details     struct {
+		} `json:"details"`
+	} `json:"payments"`
+	Refunds         []any  `json:"refunds"`
+	RequireApproval bool   `json:"require_approval"`
+	SalesChannel    string `json:"sales_channel"`
+	URL             string `json:"url"`
+	Customer        any    `json:"customer"`
+	ValidIfPending  bool   `json:"valid_if_pending"`
+}
+
+type PretixConfig struct {
+	ApiKey        string `json:"api_key"`
+	EventSlug     string `json:"event_Slug"`
+	OrganizerSlug string `json:"organizer_slug"`
+	BaseUrl       string `json:"base_url"`
+}
+
+func NewPretixConfig() *PretixConfig {
+	config := &PretixConfig{}
+	config.ApiKey = getEnv("PRETIX_API_KEY")
+	config.EventSlug = getEnv("PRETIX_EVENT_SLUG")
+	config.OrganizerSlug = getEnv("PRETIX_ORGANIZER_SLUG")
+	config.BaseUrl = getEnv("PRETIX_BASE_URL")
+
+	return config
+}
+
+var pretixConfig = NewPretixConfig()
+
+func getPretixOrder(orderCode string) (PretixOrder, error) {
+
+	var order PretixOrder
+	url := fmt.Sprintf("https://%s/api/v1/organizers/%s/events/%s/orders/%s/", pretixConfig.BaseUrl, pretixConfig.OrganizerSlug, pretixConfig.EventSlug, orderCode)
```

Comment:
This will break if any of the interpolated variables contain characters which need to be url encoded. Consider using
https://pkg.go.dev/path#Join or at least checking the constructed URL with `url.Parse()` ([example](https://pkg.go.dev/net/url#example-URL))